pkg/operator/controller/ingress/dns_test.go

```diff
@@ -109,6 +109,70 @@ func TestDesiredWildcardDNSRecord(t *testing.T) {
 	}
 }
 
+func TestDomainNotMatching(t *testing.T) {
+	tests := []struct {
+		name         string
+		domain       string
+		baseDomain   string
+		platformType configv1.PlatformType
+		expected     bool
+	}{
+		{
+			name:         "empty domain",
+			domain:       "",
+			baseDomain:   "openshift.example.com",
+			platformType: configv1.AWSPlatformType,
+			expected:     false,
+		},
+		{
+			name:         "domain matches the baseDomain",
+			domain:       "apps.openshift.example.com",
+			baseDomain:   "openshift.example.com",
+			platformType: configv1.AWSPlatformType,
+			expected:     false,
+		},
+		{
+			name:         "domain does not match the baseDomain on AWS",
+			domain:       "test.local",
+			baseDomain:   "openshift.example.com",
+			platformType: configv1.AWSPlatformType,
+			expected:     true,
+		},
+		{
+			name:         "domain does not match the baseDomain on GCP",
```

Comment:
It took me a minute to grok what this test was actually testing :smile: 

The name is a little misleading because you're not actually testing GCP as it's not currently supported for this use case. Can the name be updated to something like "unsupported platform". The API has a [NonePlatformType](https://github.com/openshift/api/blob/c689b3938fab2ff48f7735ac410e7565d6e53426/config/v1/types_infrastructure.go#L150) that you could probably use here.

Reply:
Good idea! I've changed it.